### PR TITLE
Upgrade to Realm Core v13.6.0 (on `main`)

### LINKED ISSUE #5544: Upgrade Realm Core on `main`
In the former bindgen now main branch, we're using a version of Core from Dec 5, 2022.

We should upgrade and adapt to any changes that has been introduced.

We're currently a few commits above v13.1.1 and the latest (as of 7th of March) is v13.6.0.

The comparison of changelog between the two versions: https://github.com/realm/realm-core/compare/v13.1.1...v13.6.0#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed

Our PRs on `v11` performing the upgrades:
- #5188
- #5252
- #5435
- #5453
- #5239

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This closes #5544 by following the PRs which upgraded Core on `v11` and migrating this to the new SDK.

All tests which previously passed are still passing (but there are still some outstanding test that run on `v11` which are not being imported, so we might want to fix those by picking the changes made to the tests in https://github.com/realm/realm-js/pull/5239)

### CODE REVIEW

**Comment by kraenhansen** [comment]
I feel this PR is a great testament to the bindgen project.
Of cause I had the previous PRs to lean against but adapting to breakages in the Realm Core APIs was surprisingly simple and I had a lot of help from types all the way into the SDK 🎉

**Review by kneth — APPROVED** [review]
LGTM but I wonder if `clearTestState()` needs an update (`remove_dir_recursive()` has been removed)

**Comment by kraenhansen** [comment]
I know you're out @RedBeard0531 - please review the changes to the bindgen package in retrospect.

File: packages/realm/src/platform/device-info.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+////////////////////////////////////////////////////////////////////////////
+
+import { DeviceInfo } from "../binding";
+
+type DeviceInfoType = {
+  create(): DeviceInfo;
+};
+
+export const deviceInfo: DeviceInfoType = {
+  create() {
```
Comment (takameyer) [inline packages/realm/src/platform/device-info.ts:26]: This crashes immediately on React Native now.

**Pull request merged**